xwiki-pro-macros-api/src/main/java/com/xwiki/macros/excerptinclude/internal/macro/ExcerptIncludeMacroRefactoring.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.macros.excerptinclude.internal.macro;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceResolver;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.rendering.block.MacroBlock;
+import org.xwiki.rendering.listener.reference.DocumentResourceReference;
+import org.xwiki.rendering.listener.reference.ResourceReference;
+import org.xwiki.rendering.macro.MacroRefactoring;
+import org.xwiki.rendering.macro.MacroRefactoringException;
+import org.xwiki.stability.Unstable;
+import org.xwiki.text.StringUtils;
+
+/**
+ * Implementation of reference refactoring operation for the excerpt-include macro.
+ *
+ * @version $Id$
+ * @since 1.14.6
+ */
+@Component
+@Singleton
+@Named("excerpt-include")
+@Unstable
```

Comment:
I don't think there is any need to use `@Unstable` on an internal class.

Reply:
Indeed, I removed it.